spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/pre/Servlet30WrapperFilter.java

```diff
@@ -80,6 +84,10 @@ else if (RequestUtils.isDispatcherServletRequest()) {
 			// If it's going through the dispatcher we need to buffer the body
 			ctx.setRequest(new Servlet30RequestWrapper(request));
 		}
+		else if (zuulProperties.isTraceRequestBody()) {
```

Comment:
I'm not sure this is exactly what I described.

> turn off trace request body if zuul.use-filter=true.

This enables buffering if `isTraceRequestBody()` which isn't the same.

Reply:
> turn off trace request body if zuul.use-filter=true.

What you suggest seems like it'll be really awkward to program and unexpected for users:
if `zuul.use-filter=true`, ignore `zuul.traceRequestBody` and treat it as if it were always false. This implementation would confuse users who expect `zuul.traceRequestBody` to do something, and it's weird for `traceRequestBody` to be a variable that's sometimes ignored. IMHO, making `zuul.traceRequestBody` always work as expected is a better solution for maintainability and users alike, which is the approach I took in this PR.

Reply:
It also unexpectedly introduces buffering because trace request body is true by default. I won't apply this PR as is.

Reply:
Can you provide any guidance on what I can do?
Things I can think of include making some other change that you suggest or rebasing against another branch.

Reply:
Add another case here https://github.com/spring-cloud/spring-cloud-netflix/blob/eff47e046410dd0b6f3ccd600195e0ba4f1a1827/spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java#L266

Reply:
Apologies for my density... what case are suggesting be added?

Reply:
`if zuul.use-filter=true: return false`